lib/galaxy/jobs/runners/lsf.py

```diff
@@ -0,0 +1,240 @@
+"""
+Job control via the LSF.
+"""
+import logging
+import os
+import stat
+
+from galaxy import model
+from galaxy.jobs.runners import (
+    AsynchronousJobRunner,
+    AsynchronousJobState
+)
+from galaxy.jobs.runners.util.lsf import (
+    build_submit_description,
+    lsf_stop,
+    lsf_submit,
+    submission_params,
+    lsf_bjob
+)
+from galaxy.util import asbool
+
+log = logging.getLogger(__name__)
+
+__all__ = ('LSFJobRunner', )
+
+
+class LSFJobState(AsynchronousJobState):
+    def __init__(self, **kwargs):
+        """
+        Encapsulates state related to a job that is being run via the DRM and
+        that we need to monitor.
+        """
+        super(LSFJobState, self).__init__(**kwargs)
+        self.failed = False
+        self.user_log = None
+        self.user_log_size = 0
+
+
+class LSFJobRunner(AsynchronousJobRunner):
+    """
+    Job runner backed by a finite pool of worker threads. FIFO scheduling
+    """
+    runner_name = "LSFRunner"
+
+    def __init__(self, app, nworkers):
+        """Initialize this job runner and start the monitor thread"""
+        super(LSFJobRunner, self).__init__(app, nworkers)
+        self._init_monitor_thread()
+        self._init_worker_threads()
+
+    def queue_job(self, job_wrapper):
+        """Create job script and submit it to the DRM"""
+
+        # prepare the job
+        include_metadata = asbool(job_wrapper.job_destination.params.get("embed_metadata_in_job", True))
+        if not self.prepare_job(job_wrapper, include_metadata=include_metadata):
+            return
+
+        # get configured job destination
+        job_destination = job_wrapper.job_destination
+
+        # wrapper.get_id_tag() instead of job_id for compatibility with TaskWrappers.
+        galaxy_id_tag = job_wrapper.get_id_tag()
+
+        # get destination params
+        query_params = submission_params(prefix="", **job_destination.params)
+        container = None
+        universe = query_params.get('universe', None)
+        if universe and universe.strip().lower() == 'docker':
+            container = self._find_container(job_wrapper)
+            if container:
+                # HTCondor needs the image as 'docker_image'
+                query_params.update({'docker_image': container.container_id})
+
+        galaxy_slots = query_params.get('cores', None)
+        if galaxy_slots:
+            galaxy_slots_statement = 'GALAXY_SLOTS="%s"; export GALAXY_SLOTS_CONFIGURED="1"' % galaxy_slots
+        else:
+            galaxy_slots_statement = 'GALAXY_SLOTS="1"'
```

Comment:
Does https://github.com/galaxyproject/galaxy/commit/4cdc3eb93e1438b8d796b3dec1dc912c357806fc not work for this? I'd rather populate the cores at job runtime in on the actual cluster - I think it is more robust that way.

Reply:
I see, this is in the Condor runner also. I'd try to get away without it if you can.